HHS/simpler-grants-gov
# [Issue #2696] add New Relic to Next app

## Summary
Fixes #2696

### Time to review: __15 mins__

## Changes proposed

The goal is to begin sending data to New Relic from the frontend Next application. Things that have been done:

* a New Relic entity called "Simpler Grants Next DEV" created to track Node and client / browser data
* code added as per direction from New Relic documentation and example Next app in order to add necessary fronted scripts for tracking timing data see:
    * https://www.npmjs.com/package/newrelic
    * https://github.com/newrelic/newrelic-node-examples/tree/58f760e828c45d90391bda3f66764d4420ba4990/nextjs-app-router
    * https://docs.newrelic.com/docs/apm/agents/nodejs-agent/installation-configuration/install-nodejs-agent-docker/ 
* custom typing added to plaster over deficiencies in supplied New Relic TS definitions
* support for necessary environment variables added in infra code
* tested locally as described below


## Context for reviewers

### Test steps

1. start next on this branch with `NEW_RELIC_APP_NAME="<see 1pass>" NEW_RELIC_LICENSE_KEY="<see 1pass>" npm run dev:nr`
2. visit some pages and perform searches
1. _VERIFY_: data comes through into [New Relic](https://one.newrelic.com/nr1-core/page-views/home/NTI0OTgwOXxCUk9XU0VSfEFQUExJQ0FUSU9OfDExMjAzNTAwMjc?account=5249809&duration=1800000&filters=%28domain%20IN%20%28%27BROWSER%27%2C%20%27BROWSER%27%29%20AND%20type%20IN%20%28%27APPLICATION%27%2C%20%27MICRO_FRONTEND%27%29%29&state=7a8689e5-2312-28e4-34bc-98ae1f022400), in both APM and browser dashboards

## Additional information

### Screenshots after deploying to DEV
![Screenshot 2024-11-22 at 10 04 24 AM](https://github.com/user-attachments/assets/f613349c-b60e-4d05-8a0d-35904491a134)
![Screenshot 2024-11-22 at 10 03 50 AM](https://github.com/user-attachments/assets/c41f01b9-89e1-451f-8fb2-b4e0e1d82abc)



### Next Steps

* open an issue and possibly a PR with DefinitelyTyped to fix typing issues https://github.com/DefinitelyTyped/DefinitelyTyped
* deploy to DEV to confirm functionality in deployed environment
* confirm expected behavior in PROD (data comes in) and Staging (data does not come in) after deployment
* documentation
